test/Java9andUp/src/org/openj9/test/java9AttachAPI/TargetManager.java

```diff
@@ -0,0 +1,339 @@
+	}
+
+	public BufferedReader getTargetOutReader() {
+		return targetOutReader;
+	}
+
+	public BufferedReader getTargetErrReader() {
+		return targetErrReader;
+	}
+
+	public TargetManager(String cmdName, String targetId) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, null, null, null);
+	}
+
+	/*
+	 * target must print the PID on one line, other information on following
+	 * line(s) (if any), the initialization status on the final line.
+	 */
+	public TargetStatus readTargetStatus() {
+		TargetStatus tgtStatus = TargetStatus.INIT_FAILURE;
+		StringBuffer targetLog = new StringBuffer();
+		try {
+			boolean done = false;
+			do {
+				String tgtOutput = targetOutReader.readLine();
+				if (null != tgtOutput) {
+					targetLog.append(tgtOutput);
+					targetLog.append('\n');
+				} else {
+					logger.debug("TargetVM stdout closed unexpectedly");
+					while (targetErrReader.ready()) {
+						String currentLine = targetErrReader.readLine();
+						if (null != currentLine) {
+							logger.error(currentLine + "\n");
+						}
+					}
+					tgtStatus = TargetStatus.INIT_FAILURE;
+					break;
+				}
+				if (verbose) {
+					logger.debug("TargetVM output: " + tgtOutput);
+				}
+				if (tgtOutput.startsWith(VMID_PREAMBLE)) {
+					setTargetVmid(tgtOutput.substring(VMID_PREAMBLE.length()));
+				}
+				if (tgtOutput.startsWith(STATUS_PREAMBLE)) {
+					String statusString = tgtOutput.substring(STATUS_PREAMBLE
+							.length());
+					if (statusString.equals(STATUS_INIT_FAIL)) {
+						tgtStatus = TargetStatus.INIT_FAILURE;
+					} else if (statusString.equals(STATUS_INIT_SUCESS)) {
+						tgtStatus = TargetStatus.INIT_SUCCESS;
+					}
+					done = true;
+				}
+			} while (!done);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		if (TargetStatus.INIT_SUCCESS != tgtStatus) {
+			logger.debug("TargetVM initialization failed with status "+tgtStatus.toString()+"\nTarget output:\n"+targetLog.toString());			
+		}
+		return tgtStatus;
+	}
+
+	public TargetManager(String cmdName) {
+		this.proc = launchTarget(cmdName, null, null, null, null);
+	}
+
+	public TargetManager(String cmdName, String targetId,
+			List<String> appArgs) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, null, null, appArgs);
+	}
+
+	public TargetManager(String cmdName, String targetId,
+			List<String> vmArgs, List<String> appArgs) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, null, vmArgs, appArgs);
+	}
+
+	public TargetManager(String cmdName, String targetId, String displayName,
+			List<String> vmArgs, List<String> appArgs) {
+		this.targetId = targetId;
+		this.proc = launchTarget(cmdName, targetId, displayName, vmArgs,
+				appArgs);
+	}
+
+	int waitFor() throws InterruptedException {
+		return proc.waitFor();
+	}
+
+	/**
+	 * launch a TargetVM process
+	 * 
+	 * @return launched process
+	 */
+	private Process launchTarget(String cmdName, String tgtId,
```

Comment:
This really needs some comments to walk a reader of the code through.  I'm not sure what `sidecar` is, or why this code is structured the way it is.  Aim to make it obvious for a reader (or person debugging) to understand the code - especially as that may be future you a few months down the line =)

Reply:
> especially as that may be future you a few months down the line =)
I wrote this 9 years ago :-).

Done.  Let me know if you want more.